scripts/rotate-aws-creds/common.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+aws_cli_setup() {
+    local KEY_ID=$1
+    local SECRET_KEY=$2
+    local PROFILE_NAME=$3
+    info "Using access key '$KEY_ID'."
+    echo -e "${KEY_ID}\n${SECRET_KEY}\nus-east-1\njson\n" | aws configure --profile "$PROFILE_NAME" 2>/dev/null >/dev/null
+}
+
+rotate_credential_requests() {
+    local TEMP_DIR=$1
+    local KUBECONFIG=$2
+    local OC=$3
+    
+    # Rotate other credentials by saving a copy of the CredentialsRequest, deleting the CredentialsRequest, and creating it again
+    # Note these are in core NS and NOT managed by CVO
+    # Solution: https://access.redhat.com/solutions/4279111
+    # Bug (optimization): https://issues.redhat.com/browse/CO-742
+    rm "${TEMP_DIR}/*.CredentialsRequest.json" 2>/dev/null || true
+
+    # get original credentialsrequests and delete them
+    for NS in $(KUBECONFIG=$KUBECONFIG $OC get credentialsrequests --all-namespaces -o json | jq -r '.items[].metadata.namespace' | uniq | grep -e ^openshift -e ^kube -e ^default$ -e ^redhat | grep -v openshift-cloud-credential-operator);
+    do
+        for CR in $(KUBECONFIG=$KUBECONFIG $OC get credentialsrequests -n "$NS" -o json | jq -r '.items[].metadata.name');
+        do
+            # filename to stash resource in
+            F=${TEMP_DIR}/${CR}_${NS}.CredentialsRequest.json
+
+            # get the original json and clean it up
+            KUBECONFIG=$KUBECONFIG $OC -n "$NS" get credentialsrequests "$CR" -o json | jq -r 'del(.status) | del(.metadata.annotations) | del(.metadata.creationTimestamp) | del(.metadata.finalizers) | del(.metadata.generation) | del(.metadata.ownerReferences) | del(.metadata.resourceVersion) | del(.metadata.selfLink) | del(.metadata.uid)' > "$F"
+
+            # delete the original resource
+            KUBECONFIG=$KUBECONFIG $OC -n "$NS" delete credentialsrequests "$CR"
+
+            # recreate resource
+            KUBECONFIG=$KUBECONFIG $OC create -f "$F"
+            # name of referenced secret
+            S=$(jq -r '.spec.secretRef.name' "$F")
+
+            # cleanup any pods that referenced the secret (note we're ignoring namespace, exceptionally small risk)
+            IFS_OLD=$IFS
+            IFS=
+            for X in $(KUBECONFIG=$KUBECONFIG $OC get pods --all-namespaces -o json | jq -r ".items[] | select(.spec.volumes[].secret.secretName == \"$S\") | \"$OC -n \" + .metadata.namespace + \" delete pod \" + .metadata.name");
+            do
+                X="KUBECONFIG=$KUBECONFIG $X"
+                # do not wait, just move on
+                eval "$X" &
+            done
+            IFS=$IFS_OLD
+
+            info "Rotated SRE managed CredentialsRequest '$CR' in '$NS'."
```

Comment:
You're not expecting `$CR` and `$NS` to evaluate here?

Reply:
They'll evaluate. The single quotes are literal since the whole thing is surrounded by double quotes.

```shell
[efried@efried ops-sop]$ a=foo
[efried@efried ops-sop]$ echo "this is '$a'."
this is 'foo'.
```